src/reynir_correct/checker.py

```diff
@@ -415,102 +412,25 @@ def create_sentence(self, job: Job, s: TokenList) -> Sentence:
-
-
-def check_tokens(
-    tokens: Iterable[CorrectToken], rc: Optional[GreynirCorrect] = None, **options: Any
-) -> Optional[Sentence]:
-    """Check and annotate a single sentence, given as a token list"""
-    # Returns None if no sentence was parsed
-    rc = create_rc_instance(rc, **options)
-    max_sent_tokens = options.pop("max_sent_tokens", DEFAULT_MAX_SENT_TOKENS)
-    return rc.parse_tokens(tokens, max_sent_tokens=max_sent_tokens)
-
-
-def check(text: str, rc: Optional[GreynirCorrect] = None, **options: Any) -> Iterable[Paragraph]:
-    """Return a generator of checked paragraphs of text,
-    each being a generator of checked sentences with
-    annotations"""
-    split_paragraphs = options.pop("split_paragraphs", False)
-    max_sent_tokens = options.pop("max_sent_tokens", DEFAULT_MAX_SENT_TOKENS)
-    rc = create_rc_instance(rc, **options)
-    # This is an asynchronous (on-demand) parse job
-    job = rc.submit(
-        text,
-        parse=True,
-        split_paragraphs=split_paragraphs,
-        max_sent_tokens=max_sent_tokens,
-    )
-    yield from job.paragraphs()
-
-
-def check_with_custom_parser(
-    text: str,
-    *,
-    settings: Settings,
-    parser_class: Type[GreynirCorrect] = GreynirCorrect,
-    progress_func: ProgressFunc = None,
-    split_paragraphs: bool = False,
-    annotate_unparsed_sentences: bool = True,
-    **options: Dict[str, Any],
-) -> CheckResult:
-    """Return a dict containing parsed paragraphs as well as statistics,
-    using the given correction/parser class. This is a low-level
-    function; normally check_with_stats() should be used."""
-    rc = parser_class(
-        annotate_unparsed_sentences=annotate_unparsed_sentences,
-        settings=settings,
-        **options,
-    )
-    job = rc.submit(
-        text,
-        parse=True,
-        split_paragraphs=split_paragraphs,
-        progress_func=progress_func,
-    )
-    # Enumerating through the job's paragraphs and sentences causes them
-    # to be parsed and their statistics collected
-    paragraphs = [[sent for sent in pg] for pg in job.paragraphs()]
-    return CheckResult(
-        paragraphs=paragraphs,
-        num_tokens=job.num_tokens,
-        num_sentences=job.num_sentences,
-        num_parsed=job.num_parsed,
-        ambiguity=job.ambiguity,
-        parse_time=job.parse_time,
-    )
-
-
-def check_with_stats(
-    text: str,
-    settings: Settings,
-    *,
-    split_paragraphs: bool = False,
-    progress_func: ProgressFunc = None,
-    annotate_unparsed_sentences: bool = True,
-    **options: Any,
-) -> CheckResult:
-    """Return a dict containing parsed paragraphs as well as statistics"""
-    return check_with_custom_parser(
-        text,
-        settings=settings,
-        split_paragraphs=split_paragraphs,
-        progress_func=progress_func,
-        annotate_unparsed_sentences=annotate_unparsed_sentences,
-        **options,
-    )
+    def parse_all_tokens(
```

Comment:
It looks like we're losing the asynchronous version of the grammar checking, i.e. the one that only parses and checks sentences on-demand as they are pulled out of the generator returned by check(). Is this definitely intentional?

Reply:
I've re-added the iterable parsing/correcting back into GreynirCorrect. Adding it to GreynirCorrectAPI requires more thought.